site/source/docs/contributing/developers_guide.rst

```diff
@@ -4,25 +4,46 @@
 Developer's Guide
 =================
 
-This article provides information that is relevant to people working on Emscripten itself, as opposed to those that use Emscripten in their own projects.
+This article provides information that is relevant to people that contribute
+to Emscripten. We welcome contributions from anyone that is interested to
+help out!
 
 .. tip:: The information will be less relevant if you're just using Emscripten, but may still be of interest.
 
 
-Building Emscripten from source code
-====================================
+Getting Binaries
+================
 
-As a contributor you will need to :ref:`build Emscripten from source <installing-from-source>` in order to integrate fixes back into the main repositories.
+For contributing to core Emscripten code, like ``emcc.py``, you don't need to
+build any binaries as ``emcc.py`` is in Python, and the core JS generation is
+in JavaScript. You do still need binaries for LLVM and Binaryen, which you can
+get using the emsdk,
+
+::
+
+    emsdk install tot-upstream
+    emsdk activate tot-upstream
+
+That gets a "tip-of-tree" build of the very latest binaries. You can use those
+binaries with a checkout of the core Emscripten repository, simply by calling
+``emcc.py`` from that checkout, and it will use the binaries from the emsdk.
+
+If you do want to contribute to LLVM or Binaryen, or to test modifications
+to them, you can 
+:ref:`build them from source <installing-from-source>`.
 
 Repositories and branches of interest
 =====================================
 
-The Emscripten main repository is https://github.com/emscripten-core/emscripten. There are two main branches:
+The Emscripten main repository is https://github.com/emscripten-core/emscripten.
+The main branch is **incoming**, which exists for historic reasons as we did
```

Comment:
Side note, I wonder if nowish (or perhaps the fastcomp fork) is a good time to switch to master as the tip of tree.

Reply:
I think now is as good as any time. I'd like to do it when I have the time (maybe after debug info?).